src/activity_handlers.cpp

```diff
@@ -759,7 +786,22 @@ void activity_handlers::fill_liquid_do_turn( player_activity *act_, player *p )
         case LST_INFINITE_MAP:
             // nothing, the liquid source is infinite
             break;
-        }
+        case LST_MONSTER:
+            if( source_mon->has_effect( effect_milked ) ) {
+                current_dur = source_mon->get_effect_dur( effect_milked );
+            } if( !source_mon->has_effect( effect_milked ) ) {
```

Comment:
That formating is very misleading. It *looks* like `} else if {`, but it is actually an unrelated `if`.
The `if` belong on the next line.